### ISSUE TITLE
SCEE-only quests should not affect the quest counter in SC

### ISSUE THREAD

**Issue by esilja** [Helium314/StreetComplete#691]
I see that gamification has an effect on me. I like the extra quests SCEE gives me and would like to use the app. I've however noticed that non-SC quests solved in SCEE also increase the quest counter in SC (see also #370, #645). I'd like to disable that so that
a) the quest count for SC remains comparable to other users of SC and
b) that I'm purely motivated by the extra data I enter for the SCEE-only quests since SC is more concerned about making sure the gamification doesn't lead to bad outcomes.

**Comment by mnalis** [comment]
**TL;DR:** no, I don't think it should be changed, as it would introduce problems and wouldn't really help anyway. There is no problem really, so no need for solution either :smile_cat: 

---


To elaborate:

> non-SC quests solved in SCEE also increase the quest counter in SC (see also https://github.com/Helium314/SCEE/issues/370, https://github.com/Helium314/SCEE/issues/645). I'd like to disable that 

To clarify, when you say _"I'd like to disable that so that"_  are you suggesting a change just for yourself (i.e. a preference in the settings menu), or for the whole SCEE userbase? The latter has been discussed in mentioned tickets and decided to make SCEE edits count towards total SC score (so unless there is some new information which have not been mentioned before that could change that decision, it would likely stay that way).

>  b) that I'm purely motivated by the extra data I enter for the SCEE-only quests since SC is more concerned about making sure the gamification doesn't lead to bad outcomes.

Even if you mean _"just for yourself"_; I see no solution that would make more problems than it solves, IMHO. You could fork the app and change metadata involving the word "StreetComplete" to something else so [SC statistics service](https://github.com/streetcomplete/sc-statistics-service) wouldn't recognize it as a _"SC edit"_, which would accomplish your goal, but that same _"feature"_ would introduce problems (i.e. people looking at SC notes to improve quests won't find it, fixing data  issues introduced by broken SC version won't be able to catch your changes, general [editor statistics](https://wiki.openstreetmap.org/wiki/Editor_usage_stats) won't be counting you as SC(EE) user etc).

>  a) the quest count for SC remains comparable to other users of SC and

Well, if user has tendency to _"cheat"_, it won't really be comparable anyway.  By very design, SCEE allows more edits to be made (not just by extra quests, but e.g. by raw tag editor on any node/way, etc.). 

Come to think of it, **it is not even comparable in regular SC**, esp. if one has incentive to _"score higher"_ (e.g. just by **changing _"resurvey interval"_ in settings  to _"more often"_**, it will produce up to double the number of  quest to solve, and of course you can use things/places overlay to create infinitely many POIs if you're driven purely by score). 
And **even in regular SC there are many (about 30?) quests which are disabled by default** (like all SCEE extra quests are disabled by default) that users might decide to enable if they want more quests / more opportunities to score. 

Not to mention that users in rural areas (or areas with many SC mappers) are much disadvantaged to users in urban areas with little SC mappers, even without any intention to _"score higher"_, much less cheat.   (e.g. myself mapping in the center of capital city of Zagreb might have a dozen quests in 100m radius, while someone else  in rural Adamovec might have to walk several kilometers to get even a single quest. That enormously overshadows any difference  using SC vs. SCEE might introduce)

What is measured by stars  is thus not the total effort or opportunity, but **just an effort in using the app actively**, and that stays comparable in SC vs. SCEE  -- i.e. if you solve 100 quests, it will take about the same effort of clicking[^1] in both SC and SCEE
 
Admittedly, whether you'll get presented 100 or 120 quests for the same area however is what _might_ be different depending if you use SC or SCEE, or even if use one SC vs. other SC (i.e. depending how many quests you have enabled or disabled in each), but then again, that difference is small, and will vary **much more** by other  factors e.g.  where you live), or other incentives you might have to increase score, even with correct data with less usefulness (e.g. even in SC you might decide to use Things overlay to add every single tree in a park or on a boulevard, and then additionally solve quests of about its leaf type -- while vast majority of mappers wouldn't bother doing that, so you'd build up your score quickly -- would such attention to detail be considered _"unfair score grabbing"_, though? I guess it depends on the circumstances)

What is more useful indicator than the stars itself would IMHO how often you do decide to get out and map. SC(EE) achievement "Regular" (its level), as well as separate table _"Days you've mapped recently"_, but that is another story.


---

All that being said, the very fact that you recognize yourself as being susceptible to gamification bias and thought about it, makes it very likely you won't be taking it too far. Sure, your area perhaps _might_ end up having more trees, benches and street lamps mapped than is the average case, but I trust you won't be making damage to the OSM (i.e. making stuff up)  in order to increase your score. 

Also, with time, the gamification factor of SC gets weaker (although it does not disappear completely), so there is less incentive to make edits just for the higher score.

And if you still worry about gamification of SC(EE) being a potential issue for you, might I suggest trying [Every Door](https://wiki.openstreetmap.org/wiki/Every_Door) instead? It's a little different approach (more like SC Things/Places overlay on steroids), but still very usable and notifies of things which should be rechecked if they changed; and it does not keep score. (of course, there is always [HDYC](https://wiki.openstreetmap.org/wiki/How_did_you_contribute%3F) which counts all your contributions in any OSM app, so people overly suspect to gamification might want to ignore its existence)

[^1]:  as, for example, being able to perform action on multiple object (which have been suggested previously) would unfairly reduce that scoring effort if implemented - think e.g. if one could select hundred buildings and change their building type with single click.

**Comment by esilja** [comment]
> **TL;DR:** no, I don't think it should be changed, as it would introduce problems and wouldn't really help anyway. There is no problem really, so no need for solution either 😸

I see, implementation being hard/annoying to maintain is a very fair reason to reject a feature proposal.

"there is no problem" seems wrong though, I described my problem in the initial issue text.

---

> > non-SC quests solved in SCEE also increase the quest counter in SC (see also #370, #645). I'd like to disable that
> 
> To clarify, when you say _"I'd like to disable that so that"_ are you suggesting a change just for yourself (i.e. a preference in the settings menu), or for the whole SCEE userbase?

Either would work for me.

> The latter has been discussed in mentioned tickets and decided to make SCEE edits count towards total SC score (so unless **there is some new information which have not been mentioned before** that could change that decision, it would likely stay that way).

This issue seems to be the new information which hasn't been mentioned before, a user wanting to use SCEE but trying to be motivated by the extra data being added and not the extra stars.

> Well, if user has tendency to _"cheat"_, it won't really be comparable anyway. By very design, SCEE allows more edits to be made (not just by extra quests, but e.g. by raw tag editor on any node/way, etc.).

This user (me) tries not to "cheat". The second solution in https://github.com/Helium314/SCEE/issues/370#issuecomment-1140267045 (even as a personal option in the settings) would address the problem — extra tags, raw tag editor could all change the editor name to "StreetComplete_modified" so that they don't count towards the stars in the main app.

**Comment by mnalis** [comment]
> I see, implementation being hard/annoying to maintain is a very fair reason to reject a feature proposal.

Yes. Just to clarify, it's not my call, but the available energy of maintainer and main (basically only) dev is IMHO currently likely better conserved for more important issues. 

> "there is no problem" seems wrong though, I described my problem in the initial issue text.

Sorry if it felt wrong;  the point of [TL;DR](https://en.wikipedia.org/wiki/TL%3BDR) was not to discount your issue, but to summarize conclusion (of those two pages of more detailed explanation) into just few words to minimize load on maintainer. Some nuance was [bound to be lost in the process](https://en.wikipedia.org/wiki/Phrases_from_The_Hitchhiker%27s_Guide_to_the_Galaxy#Mostly_Harmless). 


> This issue seems to be the new information which hasn't been mentioned before, a user wanting to use SCEE but trying to be motivated by the extra data being added and not the extra stars.

But _"number of stars"_ is just a counter of amount of new OSM information added (by SC/EE); so **they are the same thing**, and as such inseparable. One can decide willingly not to look at that information (or even implement technical measures which would make it harder to look up that information), but the information would still be there. There is no avoiding it.


If you added new data, your star counter (i.e. number of edits) will increase. If you didn't it won't. The SCEE does not display number of stars (i.e. number of data changes) on main screen anyway. Yes, you could willingly go to the `My Profile` statistics page of SCEE and see how much edits (i.e. how many stars) you've made and ruin your surprise, but is it that much different than going to HDYC (or your https://osm.org profile) and seeing equivalent number displayed there? It wouldn't be for me, but it might be for you. Let us know.


But after you said that you want to _"be motivated by the extra data being added and not the extra stars"_, I'm not sure what is _actually_ the issue you're having?

1. you don't want to know how much edits you've made in OSM (at all)?
2. you want to know how much edits you've made, but you don't want to know how much of them were done by SC(EE)? 
3. you want to know how much edits you've made, and you want to know how much edits were done by SC, but you don't want to know how much edits were done by disabled-by-default SC(EE) quests?
4. you want to know how much edits you've made, and you want to know how much edits were done by SC, and you want to know  how much edits were done by SC disabled-by-default quests, but you don't want to know how much edits were done by SCEE-specific disabled-by-default quests?
5. You're not opposed to knowing how much edits you've made (with SC or SCEE or other OSM editors), but you object to that number being represented by :star: emoji, as it implies gamification? You'd however be totally cool if that star icon was replaced by e.g. text "number of edits"
6. something else? (please  try to detail _exactly_ what **is** bothersome, and what **isn't**, so we can clearly understand what is your preference).


> The second solution in https://github.com/Helium314/SCEE/issues/370#issuecomment-1140267045 (even as a personal option in the settings) would address the problem

- Note that that solution requires effort in finding all such instances of string "StreetComplete" and testing what change work and what doesn't (I _think_ anything starting with "StreetComplete" won't work, but it would take several tries to find out). If you're familiar with text editing and search/replace functionality and basic GitHub usage, you can try to make those replacements, [make GitHub compile `.apk` for you](https://github.com/Helium314/SCEE/blob/modified/CONTRIBUTING_A_NEW_QUEST.md#alternative-to-android-studio) (no Android programming knowledge needed! it's just few clicks!), download it to your phone and test if it behaves to your liking. If not, continue making such search&replace changes until you reach the desired result. If successful, that will prevent scores generated from counting in any SC(EE)-app, but with all the disadvatages outlined in previous post.

- Or (significantly more easily) you can edit just this one line https://github.com/Helium314/SCEE/blob/8bc1d21e1581552634f370de4d55e9f8b0a8bffe/app/src/main/java/de/westnordost/streetcomplete/data/user/statistics/StatisticsModule.kt#L6 and change it to some invalid website (e.g. `http://example.com/notworking/i_do_not_like_statistics`) and compile that via that few `GitHub actions` clicks. That way everything will continue to work, but  your version of the SCEE won't be able to fetch the statistics.

- Alternatively (and even more easily), you can install some firewall on your Android phone (I'm using [Rethink](https://f-droid.org/en/packages/com.celzero.bravedns/) from F-droid), and configure it to block access to `westnordost.de` (where the statistics service resides). That way, you won't be able to fetch statistics and you can continue to use existing SC(EE) apps. However, that website also hosts picture Notes, so you won't be able to attach pictures to your OSM notes if you go for that approach.

Note that none of that will help you if you intentionally visit [HDYC website](https://wiki.openstreetmap.org/wiki/How_did_you_contribute%3F) or https://www.openstreetmap.org and click on your profile, as both will show how many OSM edits you've made (and HDYC will even tell you what percentage of them were made with SC/EE - i.e. _"number of stars"_).

Thus my 6 questions above, to try to find out what you actually want, and what would be the problem for you, and what wouldn't (e.g. if you can't stop yourself from going to HDYC website, _and_ being able to see those statistics is a problem for you, then only solution to not see increasing number of OSM edits would be to not contribute to OSM at all  :shrug: - but hopefully we're not talking about that)

**Comment by esilja** [comment]
Thanks for continuing to answer me and provide me with useful workarounds. This is very nice of you. I don't want to burden you with this feature request so much though, since it does seem niche, so perhaps I should just close it.

> But after you said that you want to _"be motivated by the extra data being added and not the extra stars"_, I'm not sure what is _actually_ the issue you're having?
> 
> 3. you want to know how much edits you've made, and you want to know how much edits were done by SC, but you don't want to know how much edits were done by disabled-by-default SC(EE) quests?
> 
> 4. you want to know how much edits you've made, and you want to know how much edits were done by SC, and you want to know  how much edits were done by SC disabled-by-default quests, but you don't want to know how much edits were done by SCEE-specific disabled-by-default quests?

I think 3 or 4 covers it the best. Basically, I don't like that I wouldn't be able to talk about this feature to my local friends anymore because it'd be a different number than what they have on their phones, i.e. I don't like that using SCEE interferes with the star count I see in StreetComplete.

I'm okay with seeing how many edits I made to OSM, I'd be okay with having a different SCEE star count, but it'd be cool if I could use StreetComplete normally or keep a star count similar to StreetComplete with SCEE.

Apparently I still care about the star count too much, otherwise I could just disregard it entirely and not be bothered that SCEE also adds to it.

Thanks for your time.

**Comment by Helium314** [comment]
@esilja are you using / willing to use a self-compiled version of SCEE? Changing the editor name dependent on the quest (original or SCEE) might be quite simple. Though this does not take changes in SC quests into account.

**Comment by esilja** [comment]
I'm willing to use a self-compiled version of SCEE. If you can point me to how to do the change (I'm entirely unfamiliar with the SC(EE) codebase).

**Comment by Helium314** [comment]
The simple way: change [QUESTTYPE_TAG_KEY](https://github.com/Helium314/SCEE/blob/modified/app/src/main/java/de/westnordost/streetcomplete/ApplicationConstants.kt#L8) to something like `StreetComplete_ee:quest_type`. Then none of your SCEE edits should count to statistics.

I didn't test the "full" way to ignore only SCEE-only quests, but it should be very close to:
In [`OpenChangesetsManager`](https://github.com/Helium314/SCEE/blob/modified/app/src/main/java/de/westnordost/streetcomplete/data/osm/edits/upload/changesets/OpenChangesetsManager.kt) replace `QUESTTYPE_TAG_KEY to type.name,` with
```
                if (questTypeRegistry.ordinalsAndEntries.single { it.second == type }.first >= ApplicationConstants.EE_QUEST_OFFSET)
                    "StreetComplete_ee:quest_type" to type.name
                else
                    QUESTTYPE_TAG_KEY to type.name,
```
and add a `private val questTypeRegistry: QuestTypeRegistry` in the constructor (below `private val prefs: Preferences,`).
In `QuestTypeRegistry` remove the `private` from `ordinalsAndEntries`, and in `ElementEditsModule` add another `get()` to `OpenChangesetsManager`.

**Comment by mnalis** [comment]
> The simple way: change [QUESTTYPE_TAG_KEY](https://github.com/Helium314/SCEE/blob/modified/app/src/main/java/de/westnordost/streetcomplete/ApplicationConstants.kt#L8) to something like StreetComplete_ee:quest_type. Then none of your SCEE edits should count to statistics.

Yeah I think that [should work](https://github.com/streetcomplete/sc-statistics-service/blob/a29916ffe29a7945a6a90db73850009f6a0feb14/classes/ChangesetsParser.class.php#L38) to fool score-counting.

However, one should take into consideration that doing so will fool _not only_ [sc-statistics-service](https://github.com/streetcomplete/sc-statistics-service), but also _any other data consumer_ as well -- e.g. if there is a bug with some SCEE version which causes wrong answers to be uploaded (like we had few problems in the past in SC), tools used for correcting those would not be finding errors made with SCEE which was modified in such way.
